### PR TITLE
Removes the `onUnhandledRequest` middleware option

### PR DESCRIPTION
**Pull request by baoshan**
__expected to be merged into `beta` branch__

__What?__

This PR removes the `onUnhandledRequest` middleware option.

__Why?__

To respect a proposed general rule that all Octokit middlewares (`oauth-app.js`, `webhooks.js`, `app.js`):

- do NOT handle requests outside `pathPrefix` (or `path`) option but provide a natural way for concrete runtimes/frameworks to handle unhandled requests
- always handle requests within `pathPrefix` (or `path`) option (returns `404` for unknown routes)

__But my code relies on the `onUnhandledRequest` option!__

Because `webhook.js` only exposes a (shared) middleware for Node.js and Express.js middleware currently, the natural way (to handle unhandled requests) is (AFAICT):

- For native Node.js (`http.createServer`), check returned boolean value:

```js
const middleware = createNodeMiddleware(webhooks, {});
createServer(async (req, res) => {
  if (await middleware(req, res)) return;
  res.writeHead(404);
  res.end();
}).listen();
```

- For Express.js, rely on proper handler(s) registered at your desired path(s).

-----
[View rendered README.md](https://github.com/baoshan/octokit-webhooks.js/blob/remove_on-unhandled-request/README.md)

### CODE REVIEW

**Review by wolfy1339 — APPROVED** [review]

**Comment by wolfy1339** [comment]
Test failures are unrelated to this PR.

They should have been caught in #732

**Comment by gr2m** [comment]
> Because `webhook.js` only exposes a (shared) middleware for Node.js and Express.js middleware currently, the natural way (to handle unhandled requests) is (AFAICT):
> 
>     * For native Node.js (`http.createServer`), check returned boolean value:
> 
> 
> ```js
> const middleware = createNodeMiddleware(webhooks, {});
> createServer(async (req, res) => {
>   if (await middleware(req, res)) return;
>   res.writeHead(404);
>   res.end();
> }).listen();
> ```

Should we add that to the README?

**Comment by baoshan** [comment]
Thanks! I’ve fixed that.

File: src/middleware/node/types.ts [hunk 1]
```diff
@@ -9,8 +9,4 @@ import { Logger } from "../../createLogger";
 export type MiddlewareOptions = {
   path?: string;
   log?: Logger;
-  onUnhandledRequest?: (
```
Comment (wolfy1339) [inline src/middleware/node/types.ts:12]: Can you remove the type declarations, and the associated comments that are now unused

Comment (baoshan) [reply src/middleware/node/types.ts:12]: Thanks. Done.

**Pull request merged**